### PR TITLE
[Feature] Import the automatic partition creation metadata part

### LINKED ISSUE #11444: Improve partition management
## Enhancement
Many databases support the function of importing and automatically creating partitions. We can support this function to complete the creation of partitions, and then use ttl to control the deletion of partitions. This ease of use will be better than the previous method.

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #11444

## Problem Summary(Required) ：

```
CREATE TABLE site_access(
    event_day DATE,
    site_id INT DEFAULT '10',
    city_code VARCHAR(100),
    user_name VARCHAR(32) DEFAULT '',
    pv BIGINT DEFAULT '0'
)
DUPLICATE KEY(event_day, site_id, city_code, user_name)
PARTITION BY date_trunc('month', event_day) (
START ("2015-06-01") END ("2022-12-01") EVERY (INTERVAL 1 year)
)
DISTRIBUTED BY HASH(event_day, site_id) BUCKETS 32
PROPERTIES("replication_num" = "1");
```
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 96 / 144 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|1|00.00%|[1571]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|7|28.57%|[1435, 1481, 1533, 3485, 3608]|
|:large_blue_circle:|com/starrocks/catalog/ExpressionRangePartitionInfo.java|25|53|47.17%|[104, 105, 107, 108, 110, 111, 113, 115, 116, 118, 121, 122, 123, 124, 127, 128, 129, 131, 132, 133, 135, 136, 137, 139, 140, 142, 143, 144]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|2|3|66.67%|[1355]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|30|38|78.95%|[571, 578, 581, 587, 594, 4568, 4569, 4570]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PartitionExprAnalyzer.java|12|15|80.00%|[23, 39, 40]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|7|8|87.50%|[669]|
|:large_blue_circle:|com/starrocks/sql/ast/ExpressionPartitionDesc.java|14|15|93.33%|[101]|
|:large_blue_circle:|com/starrocks/catalog/PartitionType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 96 / 144 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|1|00.00%|[1571]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|7|28.57%|[1435, 1481, 1533, 3485, 3608]|
|:large_blue_circle:|com/starrocks/catalog/ExpressionRangePartitionInfo.java|25|53|47.17%|[104, 105, 107, 108, 110, 111, 113, 115, 116, 118, 121, 122, 123, 124, 127, 128, 129, 131, 132, 133, 135, 136, 137, 139, 140, 142, 143, 144]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|2|3|66.67%|[1355]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|30|38|78.95%|[571, 578, 581, 587, 594, 4568, 4569, 4570]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PartitionExprAnalyzer.java|12|15|80.00%|[23, 39, 40]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|7|8|87.50%|[669]|
|:large_blue_circle:|com/starrocks/sql/ast/ExpressionPartitionDesc.java|14|15|93.33%|[101]|
|:large_blue_circle:|com/starrocks/catalog/PartitionType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 102 / 154 (66.23%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|2|00.00%|[1650, 1652]|
|:large_blue_circle:|com/starrocks/catalog/ExpressionRangePartitionInfo.java|25|53|47.17%|[104, 105, 107, 108, 110, 111, 113, 115, 116, 118, 121, 122, 123, 124, 127, 128, 129, 131, 132, 133, 135, 136, 137, 139, 140, 142, 143, 144]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|27|42|64.29%|[571, 578, 581, 587, 594, 602, 605, 606, 609, 4677, 4678, 4679, 4680, 4681, 4682]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PartitionExprAnalyzer.java|12|15|80.00%|[23, 39, 40]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|10|12|83.33%|[1487, 2514]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzerUtils.java|7|8|87.50%|[669]|
|:large_blue_circle:|com/starrocks/sql/ast/ExpressionPartitionDesc.java|14|15|93.33%|[101]|
|:large_blue_circle:|com/starrocks/catalog/PartitionType.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/PartitionDesc.java [hunk 1]
```diff
@@ -46,7 +46,10 @@ public String toSql() {
         throw new NotImplementedException();
     }
 
-    public PartitionInfo toPartitionInfo(List<Column> columns, Map<String, Long> partitionNameToId, boolean isTemp)
+    // Currently, RANGE is used for materialized view ExpressionRangePartitionInfo, which is isExprPartition=false,
+    // and EXPR_RANGE is used for ordinary table ExpressionRangePartitionInfo, which is isExprPartition=true
+    public PartitionInfo toPartitionInfo(List<Column> columns, Map<String, Long> partitionNameToId,
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/PartitionDesc.java:51]: you can. add another function public PartitionInfo toPartitionInfo(List<Column> columns, Map<String, Long> partitionNameToId, boolean isTemp), which call toPartitionInfo with isExprPartition to false?

File: fe/fe-core/src/main/java/com/starrocks/catalog/ExpressionRangePartitionInfo.java [hunk 2]
```diff
@@ -35,21 +49,31 @@
...
 
-    public ExpressionRangePartitionInfo(List<Expr> partitionExprs, List<Column> columns) {
+    public ExpressionRangePartitionInfo(List<Expr> partitionExprs, List<Column> columns, PartitionType type) {
         super(columns);
         Preconditions.checkState(partitionExprs != null);
         Preconditions.checkState(partitionExprs.size() > 0);
         Preconditions.checkState(partitionExprs.size() == columns.size());
         this.partitionExprs = partitionExprs;
         this.isMultiColumnPartition = partitionExprs.size() > 0;
+        this.type = type;
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/catalog/ExpressionRangePartitionInfo.java:66]: this is ambiguous. the default type is EXPR_RANGE. add some comment to explain why can change type?

**Review by ABingHuang — APPROVED** [review]
LGTM

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**